packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2665,6 +2664,10 @@ class BuildOwner {
         }
         try {
           element.rebuild();
+          assert(() {
+            _debugElementWasRebuilt(element);
```

Comment:
This is duplication with the `updateChild`'s check?

Reply:
Let `_debugElementWasRebuilt` returns true will simpler the assertion?

Reply:
No `rebuild` doesn't necessarily call `updateChild`. The widget may not have a child.

Reply:
> `rebuild` doesn't necessarily call `updateChild`. The widget may not have a child.

I mean if we check here, the new check you added to updateChild by this change is unnecessary. 
But the check above when callback is non-null should retain.

Reply:
Could you elaborate? You mean the 2 `_debugElementWasRebuilt` calls at the end of `updateChild` are not necessary?

Reply:
LayoutBuilder doesn't call rebuild when it rebuilds in `performLayout`, nor does it parent call `update` on the layout builder so those are still needed. I don't think we want to call `_debugElementWasRebuilt(context)` in `invokeLayoutCallback` since it's not guaranteed that `context` will be rebuilt.

Reply:
> LayoutBuilder doesn't call rebuild when it rebuilds in `performLayout`, nor does it parent call `update` on the layout builder so those are still needed. I don't think we want to call `_debugElementWasRebuilt(context)` in `invokeLayoutCallback` since it's not guaranteed that `context` will be rebuilt.

Add a test for this?

Reply:
We can add a test that failed #105492 to prevent regression by accident. : )